src/mscorlib/shared/System/IO/PathInternal.Windows.cs

```diff
@@ -227,6 +227,56 @@ private unsafe static int GetRootLength(char* path, int pathLength)
             return i;
         }
 
+        internal unsafe static int GetRootLength(ReadOnlySpan<char> path)
+        {
+            int pathLength = path.Length;
```

Comment:
holy code duplication batman.

Reply:
Can we just remove char* implementation For getRootLength() ?

Reply:
yes - removing `unsafe` block is general goodness